tests/conftest.py

```diff
@@ -0,0 +1,100 @@
+import os
+from contextlib import contextmanager
+
+import pytest
+from pytest_factoryboy import register
+from sqlalchemy import create_engine
+
+import snowexsql
+from snowexsql.db import db_connection_string, initialize
+from tests.factories import (
+    CampaignFactory, DOIFactory, InstrumentFactory, MeasurementTypeFactory,
+    ObserverFactory, PointDataFactory, PointObservationFactory
+)
+from .db_setup import CREDENTIAL_FILE, DB_INFO, SESSION
+
+# Make factories available to tests
+register(CampaignFactory)
+register(DOIFactory)
+register(InstrumentFactory)
+register(MeasurementTypeFactory)
+register(ObserverFactory)
+register(PointDataFactory)
+register(PointObservationFactory)
+
+
+# Add this factory to a test if you would like to debug the SQL statement
+# It will print the query from the BaseDataset.from_filter() method
+@pytest.fixture(scope='session')
+def _debug_sql_query():
+    os.environ['DEBUG_QUERY'] = '1'
+
+
+@pytest.fixture(scope='function')
+def db_test_session(monkeypatch, sqlalchemy_engine):
+    """
+    Ensure that we are using the same connection across the test suite and in
+    the API when initiating a session.
+    """
+    @contextmanager
+    def db_session(*args, **kwargs):
+        yield SESSION(), sqlalchemy_engine
+
+    monkeypatch.setattr(snowexsql.api, "db_session", db_session)
```

Comment:
This forces the session created within the API class to use the test session. It also voids the need to overwrite the DB connection string and enables the use of transactions to roll back created test data with a test function